### PR TITLE
Support CSRA head.

### PR DESCRIPTION
**Pull request by JiayuXu0**
## Motivation

[CSRA github](https://github.com/Kevinz-code/CSRA)
[CSRA paper](https://arxiv.org/abs/2108.02456)

## Modification

Add CSRA multi-label head and test case.

## Usage

```shell
python ./tools/train.py ./configs/csra/resnet101_csra_1xb16_voc07_448.py
```

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by Ezra-Yu** [comment]
Have you run experiments in the VOC2007 dataset?

**Comment by JiayuXu0** [comment]
> Have you run experiments in the VOC2007 dataset?

Thanks for your suggestion. I have added the experiments in the pull requests.

**Comment by JiayuXu0** [comment]
> Have you run experiments in the VOC2007 dataset?

I use [pre-commit hook](https://pre-commit.com/) that checks and formats codes. So re-summit codes for review. Thanks.

**Comment by JiayuXu0** [comment]
1. Add CSRA config and README.md.
2. Unified processing method with MultiLabelLinearClsHead for difficult label.

File: mmcls/models/heads/csra_head.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        else:
+            return logit
+
+    def forward_train(self, x, gt_label, **kwargs):
+        logit = 0.
+        x = self.pre_logits(x)
+        for head in self.multi_head:
+            logit += head(x)
+        gt_label = gt_label.type_as(logit)
+        # map difficult examples to positive ones
```
Comment (Ezra-Yu) [inline mmcls/models/heads/csra_head.py:86]: I lean towards adding a param `difficult_as_neg` in multi_label heads.

```
if self.difficult_as_neg:
    # xxxxx
    gt_label[gt_label < 0] = 0
else:
    # xxxxxx
    gt_label = torch.abs(gt_label)
```

**Review by Ezra-Yu — APPROVED** [review]
LGTM.

**Comment by OpenMMLab-Assistant005** [comment]
Hi @JiayuXu0 ！First of all, we want to express our gratitude for your significant PR in the MMCls project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤
